solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -690,12 +690,15 @@ public FileStore getFileStore() {
    * @see #getDefaultHttpSolrClient()
    * @see ZkController#getSolrClient()
    * @see Http2SolrClient#requestWithBaseUrl(String, String, SolrRequest)
-   * @deprecated likely to simply be moved to the ObjectCache so as to not be used
    */
-  @Deprecated
   public SolrClientCache getSolrClientCache() {
```

Comment:
No; this misses the point.  I don't like this method's existence; it's too tempting to use it when usually you shouldn't use it.

Reply:
Could you elaborate more on what the problem with this method is?   Is there a more general approach that we are supposed to take for getting clients?  at least to me, it appears super useful...   If we we have a client created, great, use it, and if not, it creates it...  

Only six classes call it, so if we want to eliminate it, it doesn't seem unsurmountable.   Just unclear on what we would replace it with.

Reply:
so `coreContainer.getDefaultHttpSolrClient().requestWithBaseUrl` already is used in some places in DistribFileStore.  However we do:

```
final var fileResponse = solrClient.requestWithBaseUrl(baseUrl, null, fileRequest);
        try (final var stream = fileResponse.getResponseStreamIfSuccessful()) {
```

can't quite figure that getResponseStreamIfSucessful to work yet